### PR TITLE
Allow e2e tests for logged-in user journeys - account users e2e tests

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- add account users e2e test
- add custom Cypress login command **allowing tests for logged-in user journeys** - all you have to do is use `cy.login({ username, password });` in your test
  - handled via puppeteer that then returns login cookies for cypress to use
  - this is a workaround - cypress doesn't allow login flows that involve external domain (cross-domain issues)
- add custom Cypress command `acceptCookiePolicy` for convenience

## QA
verify the cypress test for advantage-users has been successful: https://github.com/canonical-web-and-design/ubuntu.com/runs/3662207566

## Issue / Card

https://github.com/canonical-web-and-design/commercial-squad/issues/267

## Screenshots

image

### CODE REVIEW

File: .github/workflows/cypress.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ jobs:
       - name: Cypress run
         uses: cypress-io/github-action@v2
         with:
+          env: UBUNTU_USERNAME=peter.makowski+100@canonical.com,UBUNTU_PASSWORD=jibfdq5hmq
```
Comment (petermakowski) [inline .github/workflows/cypress.yaml:45]: Didn't manage to get github secrets being passed correctly with `cypress-io/github-action@v2`, writing this inline for now.

If there's another test user that we should use please let me know.

File: tests/cypress/support/commands.js [hunk 2]
```diff
@@ -1 +1,13 @@
 import "@testing-library/cypress/add-commands";
+
+Cypress.Commands.add("acceptCookiePolicy", () => {
```
Comment (bethcollins92) [inline tests/cypress/support/commands.js:3]: I added [this](https://github.com/canonical-web-and-design/ubuntu.com/blob/master/tests/cypress/forms_spec.js#L28) to my test but now I'm wondering if this way is better as it's more user centric?

File: .github/workflows/cypress.yaml (hunk 1 above)
Comment (bethcollins92) [reply .github/workflows/cypress.yaml:45]: I'm not too sure about this @anthonydillon might know more

File: tests/cypress/support/commands.js (hunk 2 above)
Comment (petermakowski) [reply tests/cypress/support/commands.js:3]: I'd say so.

The only reason why I'd opt for your approach would be speed, but we don't have that much to gain here.

It's important to test the cookie banner in all contexts I think. You can sometimes get bugs with z-index etc. that e2e can catch provided we're actually testing the cookie banner, not skipping it entirely.

File: .github/workflows/cypress.yaml (hunk 1 above)
Comment (anthonydillon) [reply .github/workflows/cypress.yaml:45]: Is this supposed to be a secret?

File: tests/cypress/support/commands.js (hunk 2 above)
Comment (bethcollins92) [reply tests/cypress/support/commands.js:3]: Makes sense, I'll keep that in mind for next time - thanks for clarifying

File: .github/workflows/cypress.yaml (hunk 1 above)
Comment (petermakowski) [reply .github/workflows/cypress.yaml:45]: would be nice if it was, but doesn't really have to be, it's a dummy test user

**Review by bethcollins92 — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**